### ISSUE TITLE
Unable to get streak to work

### ISSUE THREAD

**Issue by ayutheos** [justinmk/vim-sneak#135]
I might be missing something, but I can't see the difference when `let g:sneak#streak` is enabled. I get all search highlighted in magenta:

![](http://i.imgur.com/ozlDOku.png)

My settings in _vimrc:

```
let g:sneak#streak = 1
let g:sneak#use_ic_scs = 1
let g:sneak#s_next = 0

nmap <leader>s <Plug>SneakStreak
```

None of the settings above seem to work - smartcase doesn't detect upper case as shown in screenshot. Next match is selected only with `;` not `s` (which just starts another search).

I've tried unloading all my plugins but I get the same result. Using Vim 7.4 on Windows 8.

Any help appreciated.

**Comment by justinmk** [comment]
> `nmap <leader>s <Plug>SneakStreak`

Should be `nmap <leader>s <Plug>(SneakStreak)`

(Is there documentation somewhere that incorrectly suggests `<Plug>SneakStreak`?)

BTW, you don't need `let g:sneak#streak = 1` if you plan to always invoke via your `<leader>s` mapping.

> Next match is selected only with ; not s

If you want your sneak mapping to also go to the next match, use `let g:sneak#s_next = 1` (not 0). But this is determined by whatever you mapped `<Plug>Sneak_s`, not `<Plug>(SneakStreak)`.

**Comment by ayutheos** [comment]
> (Is there documentation somewhere that incorrectly suggests <Plug>SneakStreak? )

Oops, I just substituted `<Plug>SneakPrevious` with 'SneakStreak'. Should be `<Plug>(SneakStreak)`.

I've set `nmap <leader>s <Plug>(SneakStreak)` for me to see the difference between `s` and `<leader>s`.

My revised _vimrc:
```
let g:sneak#streak = 1
let g:sneak#use_ic_scs = 1
let g:sneak#s_next = 1

nmap <leader>s <Plug>(SneakStreak)
map  s <Plug>SneakNext
map  s <Plug>SneakPrevious
nmap s <Plug>Sneak_s
nmap s <Plug>Sneak_S
```

Everything seems to be the same...magenta coloured hightlights no matter streak is enabled or not, `s` starts another search, etc. However, if I press `s <enter>` while in search, it does jumps to next highlight match.

My leader is set to `,` btw.

**Comment by justinmk** [comment]
It's a reach, but what does `echo has('conceal')` return?

Does it work if you start gvim like this (from command line):

    gvim -N -u NONE -c 'let g:sneak#streak = 1' -c 'set rtp+=~/.vim/bundle/vim-sneak' -c 'runtime! plugin/sneak.vim'

**Comment by hardened-apple** [comment]
I think I have the same problem:
I've noticed that setting 'nowrap' means that sneak streak doesn't work.

**Comment by justinmk** [comment]
@hardened-apple To eliminate unknowns, does it work like this:

    vim -N -u NONE -c 'set nowrap' -c 'let g:sneak#streak = 1' -c 'set rtp+=~/.vim/bundle/vim-sneak' -c 'runtime! plugin/sneak.vim'

I use `nowrap` pretty much all the time, and occasionally switch to `wrap`, with no issues.

**Comment by hardened-apple** [comment]
@justinmk  No it doesn't work with that -- but once in the instance 'set wrap' makes everything work (similarly starting with 'wrap' and calling 'set nowrap' in the instance works at first, then stops working).

I'm pretty sure it everything's fine on my work machine, and I have the same config there, so I'm guessing something external is influencing it, but I'll have to wait until Monday to test.

**Comment by ayutheos** [comment]
> It's a reach, but what does echo has('conceal') return?

I get a `1`

Running from the windows command line:

```
C:\Windows\system32>gvim -N -u NONE -c 'let g:sneak#streak = 1' -c 'set rtp+=~/v
imfiles/bundle/vim-sneak' -c 'runtime! plugin/sneak.vim'
```
gives me:
![](http://i.imgur.com/CVmmpKw.png)

**Comment by hardened-apple** [comment]
I've found that some recent patch on vim has made my problem happen.
It was fine on my work machine as the vim version only had patches 1-567, once I upgraded with the distros package manager, so the version now has patches 1-580, the problem appears.
I'll narrow it down further once I have the time.

**Comment by justinmk** [comment]
@ayutheos I gave you the wrong syntax for windows command line, sorry. Try this instead (quotes instead of apostrophes):

```
C:\Windows\system32>gvim -N -u NONE -c "let g:sneak#streak = 1" -c "set rtp+=~/v
imfiles/bundle/vim-sneak" -c "runtime! plugin/sneak.vim"
```

@hardened-apple Very interesting, thanks for digging into that. I watch the vim_dev mailing list so I will look out for related patches.

**Comment by ayutheos** [comment]
> Try this instead (quotes instead of apostrophes):

> `C:\Windows\system32>gvim -N -u NONE -c "let g:sneak#streak = 1" -c "set rtp+=~/v
imfiles/bundle/vim-sneak" -c "runtime! plugin/sneak.vim"`

![Imgur](http://i.imgur.com/ju71PtP.png)

It works! (to a certain extent). The target labels show up (searched for `el`, but all the `e` remains highlighted even after pressing `esc`, `;`, `,` or other keys. It only goes away by entering `s <enter>`, but only on the line the cursor is on.

![Imgur](http://i.imgur.com/Ue8uK5M.png)

**Comment by hardened-apple** [comment]
@justinmk Thought I'd mention -- my distro's vim version now includes patches up to 617, and the problem's gone in that build, so the problem doesn't affect me any more.

**Comment by justinmk** [comment]
@hardened-apple Good to know, thanks! So the problem started somewhere in 7.4.568-7.4.580, and resolved somewhere in 7.4.581-7.4.617.
